### PR TITLE
New filters

### PR DESCRIPTION
**Pull request by AkkarinForest**
I added a second elm app and used it to create profile and tags filter for fixed sessions. We now have some duplication between them, I extracted part of it to Labels module, but if you have any suggestions on how to clean it further let me know. Or maybe it's ok to keep the duplication since it's temporary until we connect the elm apps together.
Also, testing gets more tricky with all these abstraction and duplications, so I'm not sure if now I cover all the stuff that should be covered, or if I'm testing some parts multiple times...

Also, if you don't want to type too much we can always have a skype session or you can record me a voice message :D

### CODE REVIEW

File: app/javascript/angular/code/controllers/_mobile_sessions_map_ctrl.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import _ from 'underscore';
-import { Elm } from '../../../elm/src/Main.elm';
+import { Elm } from '../../../elm/src/MobileSessionsFilters.elm';
```
Comment (3v0k4) [inline app/javascript/angular/code/controllers/_mobile_sessions_map_ctrl.js]: Great move renaming this too!

File: app/javascript/angular/code/controllers/_fixed_sessions_map_ctrl.js [hunk 2]
```diff
@@ -94,4 +96,61 @@ export const FixedSessionsMapCtrl = (
     }, true);
 
   $scope.setDefaults();
+
+  if (process.env.NODE_ENV !== 'test') {
```
Comment (3v0k4) [inline app/javascript/angular/code/controllers/_fixed_sessions_map_ctrl.js]: We could consider extracting the same parts into an abstraction but I'm not sure it wouldn't be wasted effort. Especially cause there's already the mobile / fixed duplication. Let's just not forget. If you want on my wall there's an "eject" section where I was collecting things to be done before wrapping up

File: app/javascript/angular/code/directives/autocomplete.js [hunk 3]
```diff
@@ -1,55 +0,0 @@
-angular.module("aircasting").directive('autocomplete', function (){
```
Comment (3v0k4) [inline app/javascript/angular/code/directives/autocomplete.js]: Great! Less Angular stuff!

File: app/javascript/elm/tests/MobileSessionsFiltersTests.elm [hunk 4]
```diff
@@ -133,53 +101,21 @@ tagsArea =
...
                     defaultModel
                         |> update (AddProfile activityValue)
                         |> Tuple.first
                         |> view
                         |> Query.fromHtml
                         |> Query.has [ Slc.text activityValue ]
-            , fuzz string "new profile label has a button" <|
-                \profileName ->
-                    { defaultModel | profiles = Labels.fromList [ profileName ] }
+            , test "new profile label has a button" <|
```
Comment (3v0k4) [inline app/javascript/elm/tests/MobileSessionsFiltersTests.elm]: Was the `fuzz` a problem on this one?

File: public/partials/usernames_and_tags.html [hunk 5]
```diff
@@ -1,30 +0,0 @@
-<div ng-controller="UsernamesAndTagsFiltersCtrl">
```
Comment (3v0k4) [inline public/partials/usernames_and_tags.html]: Awesome

File: app/javascript/elm/tests/MobileSessionsFiltersTests.elm (hunk 4 above)
Comment (AkkarinForest) [reply app/javascript/elm/tests/MobileSessionsFiltersTests.elm]: Yes, cause it sometimes produces an empty string. I think at some point I will create a custom `fuzz` that produces strings that I need and I'll revisit these tests.

**Pull request closed without merging**